### ISSUE TITLE
Notifications and Daydream

### ISSUE THREAD

**Issue by IzzySoft** [Freeyourgadget/Gadgetbridge#342]
Currently, notifications can be configured to be sent to the Pebble always, never, or "also when the screen is on". It would be nice to include an option to send them "also in Daydream" – which currently is only covered by "also when screen is on".

Background: The device might sit in its cradle for charging, which one often configures with Daydream (think of "nightstand mode"). This does not necessarily mean one watches the screen, or that the notification is "seen" – so it would be nice to have notifications sent to the "wearable" in such cases.

**Comment by danielegobbetti** [comment]
At least it looks like other apps may know when the device is in "daydream" mode, so that's a start :)

How to detect if daydream is on: http://stackoverflow.com/questions/18239955/how-to-detect-if-daydream-is-on

**Comment by IzzySoft** [comment]
Good find, @danielegobbetti – even with code available on Github :) So there are two ways it seems:

1. registering a broadcast receiver for those intents, and keep a status internally (possible problem: how to know the current state when GB is just starting up?)
1. use that [DayDreamUtils.java](https://github.com/PieterAelse/blendle/blob/master/mobile/src/main/java/com/piotapps/blendle/utils/DayDreamUtils.java) class (problem: I'm not sure if it shows the current *state* of DayDream: `isDaydreamEnabled()` IMHO just shows whether the *feature* is enabled (settings) – not if the device is actually "daydreaming". And I see no other matching method in that class.)

So maybe it's safer to ask the activity manager whether the service is running? As you know, I'm not an Android dev, so I can only give "educated guesses": [Android Check if Service is Running](https://gist.github.com/kevinmcmahon/2988931) looks like it's doing something similar. Strange that the [DreamService API ref](https://developer.android.com/reference/android/service/dreams/DreamService.html) doesn't mention anything on the topic. And my Google-Fu didn't turn up anything else useful on this topic, sorry.

**Comment by IzzySoft** [comment]
Just checking: low prio I guess – or is there any work-in-progress on this?

**Comment by cpfeiffer** [comment]
From my side, yes, low priority. Too many things on my plate already, and the maximum priority is to finish #206.

**Comment by IzzySoft** [comment]
Good point – first things first. And no changes to the roof while working on the basement :) I'll wait then for the corresponding item in the changelog before "pushing" here again. Thanks for the update!
